cardano-node/src/Cardano/Node/Configuration/TopologyP2P.hs

```diff
@@ -132,77 +138,120 @@ data LocalRootPeersGroup = LocalRootPeersGroup
   , valency    :: Int
   } deriving (Eq, Show)
 
+-- | Does not use the 'FromJSON' instance of 'RootConfig', so that
+-- 'accessPoints', 'advertise' and 'valency' fields are attached to the same
+-- object.
 instance FromJSON LocalRootPeersGroup where
   parseJSON = withObject "LocalRootPeersGroup" $ \o ->
                 LocalRootPeersGroup
-                  <$> o .: "localRoots"
+                  <$> parseJSON (Object o)
```

Comment:
I don't understand this. Won't you still end up using the `FromJSON RootConfig` instance?
`parseJSON :: FromJSON a => Value -> Parser a`

Reply:
I want to use `FromJSON RootConfig`, but I don't want to wrap it in an object, I want instead to extend it with `valency`.  The format for `LocalRootPeersGroup` is something like
```json
{ "accessPoints": ...
, "advertise": ...
, "valency": ...
}
```
First two fields are parsed with `RootConfig` instance attached to the same object as the `valency`.